### PR TITLE
Add Pokémon transparency based on IV

### PR DESCRIPTION
**Pull request by Arengo**
This PR introduces the feature to distinguish Pokémon by their IVs by making the low IV-Pokémon more transparent. 
## Description

A new toggle option is available in the Map menu, wrapped below the Pokémon marker switch. When enabling (default is ``false``) marker opacity get's weighted with Pokémon IV, ranging from 33 % for no-IV Pokémon or 0% IV Pokémon to an opacity of 100% for 100% IV Pokémon.

## Motivation and Context

This feature is introduced through requests to better distinguish Pokémon of interest on the map. Also there was a Feature Request in #1829 as well as on [DevKat](https://feedback.userreport.com/9eec75e1-1e84-4f02-b254-5e239d896805/#idea/159800).
Instead of a toggle it was also considered to introduce this as default, when ``-enc`` is enabled.

## How Has This Been Tested?

On my personal map.

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project. (Awesome JS code style)

### CODE REVIEW

File: static/js/map.js [hunk 1]
```diff
@@ -2276,8 +2283,23 @@ $(function () {
         buildSwitchChangeListener(mapData, ['gyms'], 'showGyms').bind(this)()
     })
     $('#pokemon-switch').change(function () {
+        var options = {
+            'duration': 500
+        }
+        var wrapper = $('#pokemon-wrapper')
+        if (this.checked) {
+            lastpokemon = false
```
Comment (Arengo) [inline static/js/map.js]: Not sure if this line and [L#2294](https://github.com/RocketMap/RocketMap/pull/1956/files#diff-052cfee44ae462a28560613abd7fdd14R2294) are needed for a more robust code-style. Any deeper knowledge available?

**Comment by friscoMad** [comment]
I think that no IV should be left as full opacity as there is no way for a web instance to know if the mapper instance has -enc or not, it is not great having a slider that does not do anything but and having all pokemon at 33% make it worse.
Otherwise it should be under webserver flag to disable it when it does not do anything.

**Comment by SiteEffect** [comment]
@friscoMad thought about it while implementing. Came to the conclusion for myself that either you are interested in IVs, so you only add ``-enc`` for IVs to those which you want to see, so this option is something for you to turn on **or** you don't scan for IVs at all and therefore this option could be just hidden like in #1840. This could be implemented as soon as the other PR is merged. Happy about more feedback, though :-)

**Comment by friscoMad** [comment]
If PR #1840 is merged before this and this PR is updated, then it would be perfect.
This PR is really nice, it is just the degraded UX for users of maps without encounters that bogged me.

**Comment by springjools** [comment]
I tested various tranparencies with gyms, and I noted that it's not a good indicator for a scale that is larger than say 2 or 3 steps. The reason is that you cannot see various transparencies well on the map, or how they differ. Did you also test with dark, road and light map? It has to work with all of these. 

Just my thoughts, but maybe a iv bar (like energy bar in some games) would suit better?

And similar with #1706 : would it be better to store the opacity levels in map_common.js? To be consequent...

**Comment by SiteEffect** [comment]
Regarding consistency I could move the min-opacity to map.common.js to make it easier customizable, any other value is assigned during runtime and dependent on that. Nice you mention downsides with opacity, so I did not test this on a very large scale, however when you have a crowded area, there are general distinction problems with overlapping and non-linear sized markers.

**Comment by SiteEffect** [comment]
(There are thoughts to change this PR to highlight high IV Pokémon another way.)
This PR is currently paused but might be opened in the future.

**Pull request closed without merging**